libevmasm/Assembly.cpp

```diff
@@ -74,6 +75,130 @@ unsigned Assembly::codeSize(unsigned subTagSize) const
 	}
 }
 
+void Assembly::addAssemblyItemsFromJSON(Json::Value const& _code)
+{
+	solAssert(m_items.empty());
+	solAssert(_code.isArray());
+	for (auto const& jsonItem: _code)
+		m_items.emplace_back(createAssemblyItemFromJSON(jsonItem));
+
+	for (auto current = m_items.begin(); current != m_items.end(); ++current)
+	{
+		// During the assembly json export a `JUMPDEST` is always generated after a `tag`.
+		// So we just ignore exactly these `JUMPDEST`'s.
+		auto const nextItem = next(current);
+		if (nextItem != m_items.end() &&
+			current->type() == AssemblyItemType::Tag &&
+			nextItem->type() == AssemblyItemType::Operation &&
+			nextItem->instruction() == Instruction::JUMPDEST
+		)
+			m_items.erase(nextItem);
+	}
+
+	for (auto const& item: m_items)
+		if (item.type() == AssemblyItemType::Operation && item.instruction() == Instruction::JUMPDEST)
+			throw langutil::Error(
+				1285_error,
+				langutil::Error::Type::FatalError,
+				"JUMPDEST instruction found that was not followed by tag."
+			);
```

Comment:
The error code is just a part of `langutil::Error`. So if you can throw it, it'll have the code. Just note that `solThrow()` only works with exceptions derived from the boost exception class. `langutil::Error` probably is one of these though.

But my point is that I'm not sure if throwing `langutil::Error` here is necessarily a good idea. It's a bit of an overkill because we're not even using all the nice features of our error reporting here: no source locations (because we no longer have the original JSON); only one error at a time, etc.

I think that a custom exception type with simple error messages would be good enough. It's simple enough to replace assertions with exceptions so we'll get proper error reporting but we won't have to investing too much time into it.

Reply:
I'm not using `langutil::Error` here anymore.  I just `solThrow` an `AssemblyImportException` exception.